### PR TITLE
Initial commits to add support for pizzly as a fusion caller in bcbio-CWL

### PR DESCRIPTION
**Pull request by skanwal**
As discussed - A PR with the changes so far, that we use to pull an update on development once it's merged and can be later tested using dev.

### CODE REVIEW

File: bcbio/pipeline/rnaseq.py [hunk 1]
```diff
@@ -162,6 +167,7 @@ def detect_fusions(data):
                        "dropped.")
     fusion_caller = dd.get_fusion_caller(data, [])
     if "oncofuse" in fusion_caller:
+        oncofuse_file = oncofuse.run(data)
```
Comment (roryk) [inline bcbio/pipeline/rnaseq.py]: I think this looks like a typo, oncofuse.run is called twice here.

File: bcbio/pipeline/rnaseq.py [hunk 2]
```diff
@@ -326,7 +332,7 @@ def assemble_transcripts(run_parallel, samples):
         if "stringtie" in assembler and stringtie.supports_merge(data):
             samples = run_parallel("stringtie_merge", [samples])
         else:
-            samples = run_parallel("cufflinks_merge", [samples])
+	    samples = run_parallel("cufflinks_merge", [samples])
```
Comment (roryk) [inline bcbio/pipeline/rnaseq.py]: Looks like a space got removed, so this indent block is broken.

File: bcbio/cwl/defs.py [hunk 3]
```diff
@@ -610,8 +612,14 @@ def rnaseq(samples):
             [cwlout(["summary", "multiqc"], ["File", "null"])],
             "bcbio-rnaseq", ["multiqc", "multiqc-bcbio"])]
     vc, vc_out = _variant_vc(checkpoints)
+    fusion = [s("rnaseq_fusion", "multi-parallel", 
+             [["quant", "fusion"],["reference", "fasta", "base"], dd.get_keys("gtf_file")],
+             [cwlout(["fusion", "fasta"], "File"),
+              cwlout(["fusion", "json"], "File")],
+	      "bcbio-rnaseq", ["pizzly"])]
+    #vc, vc_out = _variant_vc(checkpoints)
```
Comment (roryk) [inline bcbio/cwl/defs.py]: Cleanup the commented out code would be awesome

File: bcbio/pipeline/rnaseq.py [hunk 4]
```diff
@@ -120,6 +120,11 @@ def quantitate(data):
         data = to_single_data(kallisto.run_kallisto_rnaseq(data)[0])
         data["quant"]["tsv"] = os.path.join(data["kallisto_quant"], "abundance.tsv")
         data["quant"]["hdf5"] = os.path.join(data["kallisto_quant"], "abundance.h5")
+        data["quant"]["fusion"] = os.path.join(os.path.dirname(data["kallisto_quant"]), "fusion.txt")
```
Comment (roryk) [inline bcbio/pipeline/rnaseq.py]: Cleanup more commented out code would be great.

**Comment by skanwal** [comment]
Hi Rory,

Thanks for reviewing the commit. I have made changes as per your suggestions.
Please let me know if further changes are required.

**Comment by skanwal** [comment]
@roryk @chapmanb When you find sometime, can you please look into this PR.
Thanks.

**Comment by brainstorm** [comment]
@skanwal From what I'm seeing in your commits, there seems to be issues with indentation. I would recommend to install an editor that handles that for you (like https://code.visualstudio.com/) since, as you know, it's critical for python to get those right.

Also, before asking for a code review, make sure TravisCI stays green and clean as a whistle :)

https://travis-ci.org/bcbio/bcbio-nextgen/builds/407836764

Hope this pre-review helps!

**Comment by chapmanb** [comment]
Sehrish and Roman -- thanks for this, it's all looking great now. Sorry for the delay, I was trying to fix some other RNA-seq problems and wanted to get that sorted first so you didn't hit those errors as soon as you updated your latest-devel. I'll get it pulled and tested first thing tomorrow.

**Comment by skanwal** [comment]
Thank you so much, Brad. I was following the changes you made in the RNA-seq pipeline. Looking forward to testing those. Please let me know if there are further changes required from my side.

**Comment by vladsaveliev** [comment]
Sehrish, it's up to Brad, but it would be awesome if you replaced the tab characters with 4 spaces, following the bcbio code style.

**Comment by chapmanb** [comment]
Sehrish;
Thanks again for this, I've got it all merged and applied some post-merge fixes to the definitions and run functions to get it working for the baseline somatic.yaml test, and this is now part of the standard tests (https://github.com/bcbio/test_bcbio_cwl/commit/1243054ef40308b494e1bed0c67e6a2da58a3b03). I tried to detail all the changes in the commit but please let me know if you have any questions. As Roman and Vlad mention, it would be great in the future to use 4 spaces consistently and avoid tabs, since those cause all kinds of chaos with python. You should be able to setup your editor of choice to do this.

Thanks again for the work and this contribution and hope this keeps the process moving for full RNA-seq support in CWL.

**Comment by skanwal** [comment]
Hi Brad,

Thanks a lot for merging the PR and setting-up the standard test for it. I am now going through post-merge fixes before running the test. I'll definitely be getting in-touch if I have any question.
Would be careful in future with the spacing as well :)

**Pull request merged**